avocado/core/varianter.py

```diff
@@ -447,115 +381,66 @@ def _skip_new_data_check(self, function, args):
+        The summary/variants accepts verbosity where 0 means do not display
+        at all and maximum is up to the plugin.
+
+        :param summary: How verbose summary to output (int)
+        :param variants: How verbose list of variants to output (int)
+        :param kwargs: Other free-form arguments
         :rtype: str
         """
-        if not self.variants:
-            return ""
-        out = []
-        if summary:
-            # Log tree representation
-            out.append("Multiplex tree representation:")
-            # summary == 0 means disable, but in plugin it's brief
-            tree_repr = tree.tree_view(self.variants.root, verbose=summary - 1,
-                                       use_utf8=kwargs.get("use_utf8"))
-            out.append(tree_repr)
-            out.append("")
-
-        if variants:
-            # variants == 0 means disable, but in plugin it's brief
-            contents = variants - 1
-            out.append("Multiplex variants:")
-            for (index, tpl) in enumerate(self.variants):
-                if not self.debug:
-                    paths = ', '.join([x.path for x in tpl])
-                else:
-                    color = output.TERM_SUPPORT.LOWLIGHT
-                    cend = output.TERM_SUPPORT.ENDC
-                    paths = ', '.join(["%s%s@%s%s" % (_.name, color,
-                                                      getattr(_, 'yaml',
-                                                              "Unknown"),
-                                                      cend)
-                                       for _ in tpl])
-                out.append('%sVariant %s:    %s' % ('\n' if contents else '',
-                                                    index + 1, paths))
-                if contents:
-                    env = set()
-                    for node in tpl:
-                        for key, value in node.environment.iteritems():
-                            origin = node.environment_origin[key].path
-                            env.add(("%s:%s" % (origin, key), str(value)))
-                    if not env:
-                        continue
-                    fmt = '    %%-%ds => %%s' % max([len(_[0]) for _ in env])
-                    for record in sorted(env):
-                        out.append(fmt % record)
-        return "\n".join(out)
+        return "\n\n".join(self._variant_plugins.map_method("to_str", summary,
+                                                            variants,
+                                                            **kwargs))
 
     def get_number_of_tests(self, test_suite):
         """
         :return: overall number of tests * number of variants
         """
         # Currently number of tests is symmetrical
-        if self.variants:
-            no_variants = len(self.variants)
-            return len(test_suite) * no_variants
+        if self._no_variants:
+            return len(test_suite) * self._no_variants
```

Comment:
The thing is that this length is not really useful. People usually need the number of tests, not just number of variants. As mentioned earlier I believe we should wait till such method is useful.